contracts/cw20-ics20/src/error.rs

```diff
@@ -49,6 +49,15 @@ pub enum ContractError {
 
     #[error("Got a submessage reply with unknown id: {id}")]
     UnknownReplyId { id: u64 },
+
+    #[error("You cannot lower the gas limit for a contract on the allow list")]
+    CannotLowerGas,
+
+    #[error("Only the governance contract can do this")]
+    Unauthorized,
```

Comment:
I'd incline to adding some `(String)` with context of operation which failed etc.

I stumbled upon many such errors and each time probably you need to look for answer in specific place in code to understand it
```rust
"Only the governance contract can do this: ", {context: contract update}
```